src/KIM/kim_init.h

```diff
@@ -77,12 +81,14 @@ class KimInit : protected Pointers {
   enum model_type_enum {MO, SM};
   model_type_enum model_type;
   bool unit_conversion_mode;
+  // Pointer to KIM API Model object
+  KIM_Model *pkim;
```

Comment:
Why put this here?
There is no lifetime of a command style object, beyond the execution of the command method.

Reply:
@akohlmey `pkim` is introduced here as a member-variable and has been used in `kim_init.cpp` to being able to use the constructed object in several functions. This way, I just avoided the extra construction or defining a global variable.

Reply:
You don't need a global variable or additional construction/destruction. You also could have just declared it in the `KimInit::command()` function, set it to NULL there and then pass it as an argument to the functions that need to use it. 
I am not saying what you do is wrong (there are plenty of places in LAMMPS where similar things were done in the past), I am just trying to keep classes from accumulating needless members with any new code and make the flow of control more obvious.
When you prefer to keep it as a class member, it should be initialized to NULL in the class constructor and also set to NULL immediately after it is destroyed, so that nobody will by accident use it where it is no longer valid in future modifications. Use of class instances after being deleted has been a significant source for hard to identify bugs (since most of the time that class instance would still function with the deleted pointer address.

Reply:
I  agree with @akohlmey; this can be moved to the .cpp file and an extra argument can be added to the `kim_init()` routine.